receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -136,6 +110,38 @@ func (c *kafkaTracesConsumer) Start(_ context.Context, host component.Host) erro
 	if err != nil {
 		return err
 	}
+	if c.consumerGroup == nil {
+		saramaConfig := sarama.NewConfig()
+		saramaConfig.ClientID = c.config.ClientID
+		saramaConfig.Metadata.Full = c.config.Metadata.Full
+		saramaConfig.Metadata.Retry.Max = c.config.Metadata.Retry.Max
+		saramaConfig.Metadata.Retry.Backoff = c.config.Metadata.Retry.Backoff
+		saramaConfig.Consumer.Offsets.AutoCommit.Enable = c.config.AutoCommit.Enable
+		saramaConfig.Consumer.Offsets.AutoCommit.Interval = c.config.AutoCommit.Interval
+		if initialOffset, err := toSaramaInitialOffset(c.config.InitialOffset); err == nil {
+			saramaConfig.Consumer.Offsets.Initial = initialOffset
+		} else {
+			return err
+		}
+		if c.config.ResolveCanonicalBootstrapServersOnly {
+			saramaConfig.Net.ResolveCanonicalBootstrapServers = true
+		}
+		if c.config.ProtocolVersion != "" {
+			version, err := sarama.ParseKafkaVersion(c.config.ProtocolVersion)
+			if err != nil {
+				return err
+			}
+			saramaConfig.Version = version
+		}
+		if err := kafka.ConfigureAuthentication(c.config.Authentication, saramaConfig); err != nil {
```

Comment:
Extract all this config conversion in a separate func please.

Reply:
yes for sure. Found out that traces were not applying auth configuration this way